src/promptflow/tests/sdk_cli_test/conftest.py

```diff
@@ -144,3 +160,47 @@ def serving_client_image_python_flow(mocker: MockerFixture):
 @pytest.fixture
 def serving_client_composite_image_flow(mocker: MockerFixture):
     return create_client_by_model("python_tool_with_composite_image", mocker)
+
+
+def mock_origin(original):
+    def mock_invoke_tool(self, func, *args, **kwargs):
+        if func.__qualname__.startswith("AzureOpenAI"):
+            input_dict = {}
+            for key in kwargs:
+                input_dict[key] = kwargs[key]
+            if RecordStorage.is_replaying_mode():
+                response = RecordStorage.get_instance().get_record(input_dict)
+                return response
+
+            obj = original(self, func, *args, **kwargs)
+            if RecordStorage.is_recording_mode():
+                RecordStorage.get_instance().set_record(input_dict, obj)
+            return obj
+        return original(self, func, *args, **kwargs)
+
+    return mock_invoke_tool
+
+
+@pytest.fixture
+def recording_enabled(mocker: MockerFixture):
+    original_fun = FlowExecutionContext.invoke_tool
+    mocker.patch("promptflow._core.flow_execution_context.FlowExecutionContext.invoke_tool", mock_origin(original_fun))
+    os.environ[ENVIRON_TEST_MODE] = RecordMode.RECORD
```

Comment:
why we need to update environ here? I think it should be specified by the user before the test?

Reply:
I open "record and replay" in two tests.

To force these two tests to execute as I expected, I enforced the environment variable to enable in these two tests in current sdk/cli tests..
In future tests it is not needed since we'll enable these env vars globally.